lib/task/import/validate/csvValidatorCollection.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * Collection of CSV validator objects.
+ *
+ * @author     Steve Breker <sbreker@artefactual.com>
+ */
+class CsvValidatorCollection
+{
+    protected $validators = [];
+    protected $options = [];
+
+    protected function __construct(array $validatorArray, array $options)
+    {
+        $this->options = $options;
+
+        foreach ($validatorArray as $key => $class) {
+            $this->validators[$key] = new $class($options);
+        }
+    }
+
+    public function setOrmClasses(array $ormClasses)
+    {
+        foreach ($this->validators as $validator) {
+            $validator->setOrmClasses($ormClasses);
+        }
+    }
+
+    public function setFilename(string $filename)
+    {
+        foreach ($this->validators as $validator) {
+            $validator->setFilename($filename);
+        }
+    }
+
+    public function setColumnCount(int $longestRow)
+    {
+        foreach ($this->validators as $validator) {
+            $validator->setColumnCount($longestRow);
+        }
+    }
+
+    public function testRow(array $header, array $row)
+    {
+        foreach ($this->validators as $validator) {
+            $validator->testRow($header, $row);
+        }
+    }
+
+    public function getResultCollection(): CsvValidatorResultCollection
+    {
+        $resultCollection = new CsvValidatorResultCollection();
+
+        foreach ($this->validators as $validator) {
+            $resultCollection->appendResult($validator->getTestResult());
+        }
+
+        return $resultCollection;
+    }
+
+    /**
+     * Gets an instantiated validatorCollection object populated with
+     * the correct Validator objects for the specified $className.
+     *
+     * @param string $className             The entity name. e.g. QubitInformationObject
+     * @param array  $options               the options array passed to the validator
+     * @param string $specificValidatorList comma separated string of validator class names
+     */
+    public static function getValidatorCollection(string $className, array &$options, ?string $validatorList = null): CsvValidatorCollection
+    {
+        // Map of available validator classes.
+        $validatorsAvailable = [
```

Comment:
I think it would be better to define `$validatorsAvailable` as a (static?) class property to make it more prominent, and to make the list accessible outside this method.  An example use might be to return the list of available validators for output to the command line.

Reply:
This makes sense. I like having this map up at the top visible. Change completed.